<launch>
  <!-- Dataset Recorder Node -->
  <arg name="dodgeball_prefix" default="ball"/>
  <arg name="robot_name" default="mobile_base"/>
  <arg name="num_dodgeballs" default="2"/>
  <arg name="use_joystick" default="false"/> <!-- If false, will use keyboard -->

  <node type="process_gazebo.py" name="record_training_data"
    pkg="ml_comprobo" output="screen">
    <param name="save_filename" value="$(find ml_comprobo)/datasets/test.npy"/>
    <param name="dodgeball_prefix" value="$(arg dodgeball_prefix)"/>
    <param name="robot_name" value="$(arg robot_name)"/>
    <param name="num_dodgeballs" value="$(arg num_dodgeballs)"/>
    <param name="run_model" value="1"/>
    <param name="model_path" value="$(find ml_comprobo)/ml_models/LSTM_05_003"/>
    <param name="model_type" value="tf"/>
  </node>

  <node type="ball_spawn.py" name="ball_spawn"
    pkg="ml_comprobo" output="screen">
    <!--Specifies trajectory of balls (center, straight, neato, random)-->
    <param name="targeting" value="neato"/>

    <!--SPAWNING BOX FOR THE BALLS-->
    <param name="spawn_x_min" value="-1.5"/>
    <param name="spawn_x_max" value="1.5"/>
    <param name="spawn_y_min" value="5"/>
    <param name="spawn_y_max" value="5"/>
    <!--VELOCITY RANGE FOR BALLS-->
    <param name="a_vel_min" value="0"/>
    <param name="a_vel_max" value="0"/>
    <param name="l_vel_min" value="3"/>
    <param name="l_vel_max" value="5"/>
  </node>
  <!-- Custom Teleop Node -->
  <node type="training_controller.py" name="training_controller"
    pkg="ml_comprobo" output="screen"/>

  <!-- RVIZ Node -->
  <!-- <node type="rviz" name="rviz" pkg="rviz"
    args="-d $(find ml_comprobo)/rviz/config.rviz" /> -->

</launch>
